CI does not properly handle lint errors

**Describe the issue**
The `lint-server` github action https://github.com/elyra-ai/elyra/blob/main/.github/workflows/build.yml#L46-L52 does not terminate as expected if `flake8` raises an error, as seen [here](https://github.com/elyra-ai/elyra/runs/7811501848?check_suite_focus=true) and all subsequent CI runs.

Note that `flake8` returns a non-zero status code, yet the next commands (echo + black) in the `Makefile` are executed: 
![image](https://user-images.githubusercontent.com/13068832/184819394-4aa33a2b-204e-4b73-ac6b-457758a733ea.png)
Since the last command (`python -m black ...`) returns a zero return code, the action's outcome is interpreted as success.

`make lint-server` behaves as expected in a local build environment:

```
$ make lint-server
python3 -m flake8 elyra .github
elyra/tests/pipeline/test_handlers.py:33:1: I100 Import statements are in the wrong order. 'from elyra.pipeline.processor import PipelineProcessorManager' should be before 'from elyra.tests.util.handlers_utils import expected_http_error'
elyra/tests/pipeline/test_handlers.py:33:1: I202 Additional newline in a group of imports. 'from elyra.pipeline.processor import PipelineProcessorManager' is identified as Application and 'from elyra.tests.util.handlers_utils import expected_http_error' is identified as Application.
elyra/tests/pipeline/test_handlers.py:36:1: I100 Import statements are in the wrong order. 'from elyra.pipeline.parser import PipelineParser' should be before 'from elyra.pipeline.validation import ValidationResponse'
make: *** [lint-server] Error 1

```

This appears to be a `make` version issue.  If I use the latest `make`:
```
$ /usr/local/Cellar/make/4.3/bin/gmake -v         
GNU Make 4.3
Built for x86_64-apple-darwin21.1.0
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

then run it directly against the `lint-server` target, I get the following:
```
$ /usr/local/Cellar/make/4.3/bin/gmake lint-server
flake8 elyra .github
echo python3 -m black --check --diff --color .
python3 -m black --check --diff --color . || (echo "Black formatting encountered issues.  Use 'make black-format' to apply the suggested changes."; exit 1)
elyra/tests/pipeline/test_handlers.py:33:1: I100 Import statements are in the wrong order. 'from elyra.pipeline.processor import PipelineProcessorManager' should be before 'from elyra.tests.util.handlers_utils import expected_http_error'
elyra/tests/pipeline/test_handlers.py:33:1: I202 Additional newline in a group of imports. 'from elyra.pipeline.processor import PipelineProcessorManager' is identified as Application and 'from elyra.tests.util.handlers_utils import expected_http_error' is identified as Application.
elyra/tests/pipeline/test_handlers.py:36:1: I100 Import statements are in the wrong order. 'from elyra.pipeline.parser import PipelineParser' should be before 'from elyra.pipeline.validation import ValidationResponse'
python3 -m black --check --diff --color .
Skipping .ipynb files as Jupyter dependencies are not installed.
You can fix this by running ``pip install black[jupyter]``
All done! ✨ 🍰 ✨
141 files would be left unchanged.
```

So we know `make >= 4.2.1` exhibit this behavior, while `GNU make == 3.81` does not.  I suspect this make be a version 4.0 change (sigh).

--- a/Makefile
+++ b/Makefile
@@ -25,8 +25,6 @@
 .PHONY: elyra-image elyra-image-env publish-elyra-image kf-notebook-image publish-kf-notebook-image
 .PHONY: container-images publish-container-images validate-runtime-image validate-runtime-images
 
-.ONESHELL:
-
 SHELL:=/bin/bash
 
 # Container execs

